zcash_client_sqlite/src/lib.rs

```diff
@@ -1396,7 +1446,39 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                             None,
                         )?;
                     }
-                    _ => (),
+                    #[cfg(feature = "transparent-inputs")]
+                    Recipient::EphemeralTransparent {
+                        receiving_account,
+                        ephemeral_address,
+                        outpoint_metadata,
+                    } => {
+                        wallet::transparent::put_transparent_output(
+                            wdb.conn.0,
+                            &wdb.params,
+                            outpoint_metadata,
+                            &TxOut {
+                                value: output.value(),
+                                script_pubkey: ephemeral_address.script(),
+                            },
+                            None,
+                            ephemeral_address,
+                            *receiving_account,
+                        )?;
+                        wallet::transparent::ephemeral::mark_ephemeral_address_as_used(
+                            wdb,
+                            ephemeral_address,
+                            tx_ref,
+                        )?;
+                    }
+                    #[cfg(feature = "transparent-inputs")]
+                    Recipient::External(zcash_address, PoolType::Transparent) => {
+                        // Always reject sending to one of our ephemeral addresses.
+                        wallet::transparent::ephemeral::check_address_is_not_ephemeral(
+                            wdb,
+                            &zcash_address.encode(),
+                        )?;
```

Comment:
blocking: This isn't the right place for this check; it should be in transaction construction. This check becomes part of the contract that `store_sent_tx` must implement if it's done here, it could result in the store operation failing after the transaction has already been sent to the chain.

Reply:
You were right and this is fixed.